src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentResource.java

```diff
@@ -270,12 +273,23 @@ public TextAssessmentResource(AuthorizationCheckService authCheckService, TextAs
         final TextSubmission textSubmission = optionalTextSubmission.get();
         final Participation participation = textSubmission.getParticipation();
         final TextExercise exercise = (TextExercise) participation.getExercise();
-        Result result = textSubmission.getResultForCorrectionRound(correctionRound);
-
         final User user = userRepository.getUserWithGroupsAndAuthorities();
         checkAuthorization(exercise, user);
         final boolean isAtLeastInstructorForExercise = authCheckService.isAtLeastInstructorForExercise(exercise, user);
 
+        if (!authCheckService.isAtLeastTeachingAssistantForExercise(exercise, user) || resultId > 0 && !authCheckService.isAtLeastInstructorForExercise(exercise, user)) {
```

Comment:
This can use `isAtLeastInstructorForExercise` instead of calling the authCheckService for the last part of the if.